app/lib/search/search_service.dart

```diff
@@ -371,6 +377,26 @@ class SearchQuery {
     return values.isEmpty ? null : values.first;
   }
 
+  /// Returns the validity status of the query.
+  QueryValidity evaluateValidity() {
+    // Block search on unreasonably long search queries (when the free-form
+    // text part is longer than one would enter via the search input field).
+    final queryLength = parsedQuery?.text?.length ?? 0;
+    if (queryLength > _maxQueryLength) {
+      return QueryValidity.reject(rejectReason: 'Query too long.');
+    }
+
+    // Do not allow override of required tags. (E.g. search scope would require
+    // sdk:flutter, do not allow -sdk:flutter to override it).
+    final invalidOverride =
+        tagsPredicate._firstInvalidOverride(parsedQuery.tagsPredicate);
+    if (invalidOverride != null) {
+      return QueryValidity.reject(
+          rejectReason: 'Invalid override: `$invalidOverride`.');
```

Comment:
Can't we change this?

Either say `invalid tag`? or something else...

Ideally, we could explain that: You need to search from another context or change "advanced options" to avoid this conflict.

Maybe:
> The tag `sdk:flutter` is given from search filters and cannot be combined with `-sdk:flutter`. 

Or:
> The tag `-sdk:flutter` conflicts with search filters.